src/components/hooks/useAsyncRequest.tsx

```diff
@@ -0,0 +1,26 @@
+import { useCallback, useState } from 'react';
+import { useDispatch } from 'react-redux';
+
+export const useAsyncRequest = (action: Function) => {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState('');
+  const dispatch = useDispatch();
+  const makeRequest = useCallback(
+    async (data?: any) => {
+      try {
+        setError('');
```

Comment:
Why do you think that error is a simple string? What if you will get some structure with error fields and descriptions?